scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala

```diff
@@ -29,16 +29,26 @@ object JSFacade {
     }.asInstanceOf[JSLong]
   }
 
+  private[this] def toNode(t: Any): js.Any = t match {
+    case t: Tree => toNode(t)
+    case tt: Seq[_] => tt.map(toNode).toJSArray
+    case t: Option[_] => t.map(toNode).orUndefined
+    case _ => ()
+  }
+
   private[this] def toNode(t: Tree): js.Dynamic = {
     val base = js.Dynamic.literal(
       "type" -> t.productPrefix,
-      "children" -> t.children.map(toNode).toJSArray,
```

Comment:
It could be something as a method that takes a json tree and filters out the `type` and `pos` fields and flattens lists. You think that would be useful? We can skip it otherwise.